### PR TITLE
feat(css/ast): union all at-rules to one at-rule in AST

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Union all at-rules in one `AtRule` structure.

Also:
- fix bugs in codegen with printing vendor at-rules (before we print `@--webkit-keyframes` as `@keyframes`, but it is invalid because developer want to print prefixes at-rules)
- fix bug with some lints rules + TODO for future improvement
- fix bubs in minifier
- more tests
- refactor and simplify code
- parse new `@font-paletter-values` at-rule

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_ast/src/at_rule.rs [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+pub struct MediaQueryList {
+    pub span: Span,
+    pub queries: Vec<MediaQuery>,
+}
+
+#[ast_node("MediaQuery")]
+pub struct MediaQuery {
+    pub span: Span,
+    pub modifier: Option<Ident>,
+    pub media_type: Option<Ident>,
```
Comment (kdy1) [inline crates/swc_css_ast/src/at_rule.rs:169]: I think we need `#[serde(rename = "mediaType")]`

File: crates/swc_css_ast/src/at_rule.rs [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+pub enum AtRulePrelude {
+    #[tag("ListOfComponentValues")]
+    ListOfComponentValues(ListOfComponentValues),
+    #[tag("Str")]
+    CharsetPrelude(Str),
+    #[tag("CustomPropertyName")]
+    PropertyPrelude(CustomPropertyName),
+    #[tag("CustomIdent")]
+    CounterStylePrelude(CustomIdent),
+    #[tag("ColorProfileName")]
```
Comment (kdy1) [inline crates/swc_css_ast/src/at_rule.rs:36]: This is wrong.

Comment (kdy1) [reply crates/swc_css_ast/src/at_rule.rs:36]: It should be

```rust
#[tag("DashedIdent")]
#[tag("Ident")]
```

instead

File: crates/swc_css_ast/src/at_rule.rs [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+    #[tag("DashedIdent")]
+    DashedIdent(DashedIdent),
+    #[tag("Ident")]
+    Ident(Ident),
+}
+
+#[ast_node("DocumentPrelude")]
+pub struct DocumentPrelude {
+    pub span: Span,
+    pub matching_functions: Vec<DocumentPreludeMatchingFunction>,
```
Comment (kdy1) [inline crates/swc_css_ast/src/at_rule.rs:77]: I found one more renaming issue

File: crates/swc_css_ast/src/at_rule.rs [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+    CounterStylePrelude(CustomIdent),
+    #[tag("ColorProfileName")]
+    ColorProfilePrelude(ColorProfileName),
+    #[tag("DocumentPrelude")]
+    DocumentPrelude(DocumentPrelude),
+    #[tag("DashedIdent")]
+    FontPaletteValuesPrelude(DashedIdent),
+    #[tag("SelectorList")]
+    NestPrelude(SelectorList),
+    #[tag("KeyframesName")]
```
Comment (kdy1) [inline crates/swc_css_ast/src/at_rule.rs:44]: Same here, it should be two tag

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — DISMISSED** [review]
Thank you!


---

swc-bump:
 - swc_css_ast --breaking

**Comment by kdy1** [comment]
Can you fix clippy?
https://github.com/swc-project/swc/runs/5774452396?check_suite_focus=true

**Comment by alexander-akait** [comment]
WIP on clippy

**Pull request merged**